### PR TITLE
Search Engine project

### PR DESCRIPTION
**Pull request by JimmyRomero**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.fundacionjala.kata</groupId>
+    <artifactId>kata</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+
```
Comment (carledriss) [inline pom.xml:18]: Where is junit dependency?

File: src/test/java/WildsTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * WildsTest test class
+ */
+public class WildsTest {
+
+    String haystack = "Once upon a midnight dreary, while I pondered, weak and weary";
```
Comment (carledriss) [inline src/test/java/WildsTest.java]: This should be a constant.
private static final

File: src/main/java/SearchEngine.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.regex.Pattern;
+
+/**
+ * Created by JimmyRomero on 9/2/2016.
+ */
+public class SearchEngine {
+
+    static int find(String needle, String haystack) {
+
+        String regexNeedle = "\\Q" + needle.replaceAll("_", "\\\\E.\\\\Q") + "\\E";
```
Comment (carledriss) [inline src/main/java/SearchEngine.java]: this should be final

File: src/main/java/SearchEngine.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        Matcher matcher = Pattern.compile(regexNeedle).matcher(haystack);
+        if (matcher.find()) {
```
Comment (carledriss) [inline src/main/java/SearchEngine.java]: You can improve this.

``` java
return matcher.find() ? matcher.start() : -1;
```

**Comment by carledriss** [comment]
@JimmyRomero 
Let me know when you have fixed the observations.

**Comment by JimmyRomero** [comment]
@carledriss 
ready to code review

**Comment by carledriss** [comment]
Looks good to me!

**Pull request merged**